cpp/src/arrow/compute/meson.build

```diff
@@ -0,0 +1,119 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# Meson does not allow you to glob for headers to install. See also
+# https://mesonbuild.com/FAQ.html#why-cant-i-specify-target-files-with-a-wildcard
+# install_subdir would be usable if the directory only contained headers
+
+install_headers(
+    [
+        'api.h',
+        'api_scalar.h',
+        'api_vector.h',
+        'cast.h',
+        'exec.h',
+        'expression.h',
+        'function.h',
+        'function_options.h',
+        'kernel.h',
+        'ordering.h',
+        'registry.h',
+        'type_fwd.h',
+        'util.h',
+    ],
+    subdir: 'arrow/compute',
+)
+
+if needs_compute
+    pkg.generate(
+        filebase: 'arrow-compute',
+        name: 'Apache Arrow Compute',
+        description: 'All compute kernels for Apache Arrow',
+        requires: ['arrow'],
+    )
+endif
+
+arrow_compute_testing_srcs = []
+if needs_testing
+    arrow_compute_testing_srcs += files('test_util_internal.cc')
```

Comment:
As an FYI I wrapped these file paths with the `files` function because both the `compute` and `compute/kernels` configuration would reference this variable. The `files` function makes it so that the path here remains absolute, whereas without it usage of this variable in the `compute/kernels` configuration would refer to the `test_util_internal.cc` module within that directory.

It may be a good idea to generally wrap all file paths with this function